astropy/cosmology/io/tests/test_html.py

```diff
@@ -0,0 +1,254 @@
+        """Tests renaming columns"""
+        fp = tmp_path / "test_rename_html_table_columns.html"
+
+        write(fp, format="ascii.html")
+
+        tbl = QTable.read(fp)
+
+        # asserts each column name has not been reverted yet
+        for column_name in tbl.colnames[2:]:
+            # for now now, Cosmology as metadata and name is stored in first 2 slots
+            assert column_name in _FORMAT_TABLE.values()
+
+        cosmo = read(fp, format="ascii.html")
+        converted_tbl = cosmo.to_format("astropy.table")
+
+        # asserts each column name has been reverted
+        for column_name in converted_tbl.colnames[1:]:
+            # for now now, metadata is still stored in first slot
+            assert column_name in _FORMAT_TABLE.keys()
+
+    @pytest.mark.skipif(not HAS_BS4, reason="requires beautifulsoup4")
+    def test_readwrite_html_subclass_partial_info(self, cosmo_cls, cosmo, read,
+                                                  write, tmp_path, add_cu):
+        """
+        Test writing from an instance and reading from that class.
+        This works with missing information.
+        """
+        fp = tmp_path / "test_read_html_subclass_partial_info.html"
+
+        # test write
+        write(fp, format="ascii.html")
+
+        # partial information
+        tbl = QTable.read(fp)
+
+        # tbl.meta.pop("cosmology", None) # metadata not implemented
+        del tbl["$$T_{0}$$"]  # format is not converted to original units
+        tbl.write(fp, overwrite=True)
+
+        # read with the same class that wrote fills in the missing info with
+        # the default value
+        got = cosmo_cls.read(fp, format="ascii.html")
+        got2 = read(fp, format="ascii.html", cosmology=cosmo_cls)
+        got3 = read(fp, format="ascii.html", cosmology=cosmo_cls.__qualname__)
+
+        assert (got == got2) and (got2 == got3)  # internal consistency
+
+        # not equal, because Tcmb0 is changed, which also changes m_nu
+        assert got != cosmo
+        assert got.Tcmb0 == cosmo_cls._init_signature.parameters["Tcmb0"].default
+        assert got.clone(name=cosmo.name, Tcmb0=cosmo.Tcmb0, m_nu=cosmo.m_nu) == cosmo
+        # but the metadata is the same
+        # assert got.meta == cosmo.meta # metadata read not implemented
+
+    @pytest.mark.skipif(not HAS_BS4, reason="requires beautifulsoup4")
+    def test_readwrite_html_mutlirow(self, cosmo, read, write, tmp_path, add_cu):
+        """Test if table has multiple rows."""
+        fp = tmp_path / "test_readwrite_html_mutlirow.html"
+
+        # Make
+        cosmo1 = cosmo.clone(name="row 0")
+        cosmo2 = cosmo.clone(name="row 2")
+        table = vstack([c.to_format("astropy.table") for c in (cosmo1, cosmo, cosmo2)],
+                       metadata_conflicts='silent')
+
+        cosmo_cls = type(cosmo)
+        if cosmo_cls == NoneType:
+            assert 1 == 0
+
+        for n, col in zip(table.colnames, table.itercols()):

```

Comment:
Again, I'd prefer `name` -- years of training to immediately see `n` as an integer.

Reply:
Also, why have the same code reproduced here? That does not seem much of a test...

Reply:
It's reproduced here because ``write_html_table`` only works on a single cosmology and this test is for reading a multi-row table. This can be changed when Cosmology becomes array-valued.